std/math.d

```diff
@@ -1322,39 +1322,45 @@ float cosh(float x) @safe pure nothrow @nogc  { return cosh(cast(real) x); }
  *      $(TR $(TD $(PLUSMN)$(INFIN)) $(TD $(PLUSMN)$(INFIN)) $(TD no))
  *      )
  */
-real sinh(real x) @safe pure nothrow @nogc
-{
-    //  sinh(x) =  (exp(x)-exp(-x))/2;
-    // Very large arguments could cause an overflow, but
-    // the maximum value of x for which exp(x) + exp(-x)) != exp(x)
-    // is x = 0.5 * (real.mant_dig) * LN2. // = 22.1807 for real80.
-    if (fabs(x) > real.mant_dig * LN2)
-    {
-        return copysign(0.5 * exp(fabs(x)), x);
-    }
-
-    const real y = expm1(x);
-    return 0.5 * y / (y+1) * (y+2);
-}
+real sinh(real x) @safe pure nothrow @nogc { return _sinh(x); }
 
 /// ditto
-double sinh(double x) @safe pure nothrow @nogc { return sinh(cast(real) x); }
+double sinh(double x) @safe pure nothrow @nogc { return _sinh(x); }
 
 /// ditto
-float sinh(float x) @safe pure nothrow @nogc  { return sinh(cast(real) x); }
+float sinh(float x) @safe pure nothrow @nogc { return _sinh(x); }
 
 ///
 @safe unittest
 {
-    assert(isIdentical(sinh(0.0), 0.0));
-    assert(sinh(1.0).approxEqual((E - 1.0 / E) / 2));
+    enum sinh1 = (E - 1.0 / E) / 2;
+    import std.meta : AliasSeq;
+    static foreach (F; AliasSeq!(float, double, real))
+    {
+        assert(isIdentical(sinh(F(0.0)), F(0.0)));
+        assert(sinh(F(1.0)).approxEqual(F(sinh1)));
+    }
 }
 
-@safe @nogc nothrow unittest
+private F _sinh(F)(F x)
```

Comment:
Similar functions have just gone with the naming convention `sinhImpl`, but not bothered either way.